modules/internal/ChapelArray.chpl

```diff
@@ -1733,15 +1744,80 @@ module ChapelArray {
        instance of ``val`` in the array, or if ``val`` is not found, a
        tuple containing ``false`` and an unspecified value is returned.
      */
-     @unstable "'Array.find' is unstable"
-     proc find(val: this.eltType): (bool, index(this.domain)) {
+     deprecated "The tuple-returning version of `.find() on arrays` is deprecated; to opt into the new index-returning version, recompile with `-suseNewArrayFind`"
+     proc find(val: this.eltType): (bool, index(this.domain)) where !useNewArrayFind {
       for i in this.domain {
         if this[i] == val then return (true, i);
       }
       var arbInd: index(this.domain);
       return (false, arbInd);
     }
 
+
+    /*
```

Comment:
I'd add a sentence indicating what happens if `val` appears in the array multiple times. I'm assuming it returns the first one according to the iteration order of the domain right?